File: 16A.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
	int n,m;
	cin>>n>>m;
	char x[n][m];
	for(int i=0;i<n;i++)
	{
		cin>>x[i][0];
		if(x[i][0] == x[i-1][0]){
			cout<<"NO";
			exit(0);
		}
		for(int j=1;j<m;j++)
		{
			cin>>x[i][j];
			if(x[i][j] != x[i][j-1])
			{
				cout<<"NO";
				exit(0);
			}
		}
	}
	cout<<"YES";
}

File: 16A.py
n,m = map(int, raw_input().split(" "))
x=[str(raw_input())]
for i in range(1, n):
	m = str(raw_input())
	if m != x[i-1]:
		x.append(str(raw_input()))
		for i in x:
			if i != x[0]:
				print "NO"
				exit(0)
	else:
		print "NO"
		exit(0)
print "YES"